### PR TITLE
[1.10.2]Compressed Block registry rework

### PR DESCRIPTION
**Pull request by Antifluxfield**
Why you are registering those compressed blocks with the algorithm highly depending on the order of the materials?

### CODE REVIEW

**Review by Exidex — CHANGES_REQUESTED** [review]
Fix your whitespaces and formatting, also there is RegistryNamespaced#getIDForObject method that does exactly the same as the one you added

**Comment by Antifluxfield** [comment]
Done.

**Review by Exidex — COMMENTED** [review]
So, going to the point. I am not sure if this change is necessary. Existing code and proposed, both have a requirement not to change material ids order. The only difference I see is if we'll want to make material 'not ignored' (I don't think will want to do that) or change a material type, this will not break backward compatibility, but this is at the cost of increasing amount of blocks created. Also this code will change in 1.13 because of vanilla changes.

**Comment by Antifluxfield** [comment]
You should know that the order of the materials are always more volatile than the fixed ids. Imagine someone register his own material through the configs or by an addon:
Suppose our current materials is occupying the id segment O, an addon registered its materials to id segment A, and we append some materials once in an update in id segment B. Assume that A is following B, If a player takes the addon before taking the update, the order of the materials will change form "O->A" to "O->B->A", and breaking the backward compatibility. Same thing will happen if A lays before B and the player takes the addon after the update.
The only way to solve this issue is to patch a config to adjust the ids for each case when you want to append some material, and let the players/modpackers to set them to a proper value - I don't think that's a good idea since we can fix it know.

File: src/main/java/gregtech/common/blocks/MetaBlocks.java [hunk 1]
```diff
@@ -106,36 +107,40 @@ public static void init() {
...
         int generationIndex = 0;
         for(Material material : Material.MATERIAL_REGISTRY.getObjectsWithIds()) {
             if(material instanceof DustMaterial) {
-                materialBuffer.add((DustMaterial) material);
-                if(materialBuffer.size() == 16) {
-                    createCompressedBlock(materialBuffer, generationIndex);
-                    materialBuffer.clear();
-                    generationIndex++;
-                }
+            	int id = Material.MATERIAL_REGISTRY.getIDForObject(material), index = id / 16;
```
Comment (Exidex) [inline src/main/java/gregtech/common/blocks/MetaBlocks.java]: I spend some time looking where index was declared. Separate this line into two and we are good to merge

**Review by Exidex — CHANGES_REQUESTED** [review]

**Pull request merged**